### ISSUE TITLE
poloniex.ETH-BTC No trader API key provided. Starting in advisor mode. --Zen

### ISSUE THREAD

**Issue by DrBlackross** [DeviaVir/zenbot#110]
i know i have my key and secret set up

var c = module.exports = require('./config_defaults')()

// mongo stuff
c.mongo_url = "mongodb://localhost:27017/zenbrain" // change if your mongo server isn't local
c.mongo_username = null // normally not needed
c.mongo_password = null

c.poloniex_key = 'funky chicken magic here' // TO ENABLE BOT TRADING: set this to GDAX api key,
c.poloniex_secret = 'that chicken egg here' // set this to GDAX api secret,
// c.gdax_passphrase = '' // set this to GDAX api passphrase.
c.trade_log = true // log new trades as they come in.

// watch these exchanges
c.watch_exchanges = [
  //"bitfinex",
  //"gdax",
  //"kraken",
  "poloniex"
]

// selector for indicators, trading, etc
c.default_selector = "poloniex.BTC-ETH"

// add selectors in the format "{exchange-slug}.{asset}-{currency}" to graph them
c.graph_selectors = [
  c.default_selector,
  "poloniex.ETH-BTC",
  "poloniex.ETH-USD"
]

// trade logic
c.logic = require('./default_logic')

runs and will not pull a balance or try to trade..... 

OK, out of the box 

./zenbot launch map --backfill reduce run server --config config_eth_btc.js

the above works fine 
switched over to poloniex, changed all the gdax over to poloniex, fires up... loads the webpage data right. but now my api fails, should i issue/generate a new one?

all i want is to bounce back and forth between btc and eth for now, i would love is if it will pull all my balances and decide which altcoin to convert to btc, then repeat the btc eth loop

**Comment by DrBlackross** [comment]
also default_logic.js has gdax in there well... going to find replace with poloniex

**Comment by deadheadtn** [comment]
check the PULL #79 you'll find the poloniex_logic.js ready

**Comment by DrBlackross** [comment]
ugh, git is too hard to follow at my age, so....

from
https://github.com/carlos8f/zenbot/pull/79/commits/04db9614cd740e2aafb11d7f3c0d6bf9597dae75#diff-c160dd924cc2f27663c69bf6723b7754

create a new directory in "core" called "commands" and put/create all those
files in there?

correct? i can't just change branches and pull this, right?
(that is if i am reading that page right...)

**Comment by DrBlackross** [comment]
n/m got it

**Comment by DrBlackross** [comment]
ok i give i just cant get it to work....

02/05/2017 07:08:49 PM EST [    poloniex]    m24772328       2 trades. 02/05/2017 07:08:32 PM EST  BUY         0.049 at    0.01101556 ETH/BTC
02/05/2017 07:09:01 PM EST [      trader] poloniex.ETH-BTC No trader API key provided. Starting in advisor mode. --Zen
02/05/2017 07:09:01 PM EST [      trader] poloniex.ETH-BTC RSI:23 trend: null -> DOWN
02/05/2017 07:09:01 PM EST [      trader] poloniex.ETH-BTC skipping historical tick zb:m24772328

everything works fine in gekko just fine.... thing is i can't see the data like i can in zenbot

**Comment by rondijkstra** [comment]
Hello. I have got the simulation working. But once I start with the real trading it gives me an error: 
`Error: please configure your Poloniex credentials in /Users/ron/zenbot/extensions/poloniex/conf.js
`

Now, this conf.js wasn't existing, so I copied that from the zenbot 'root' directory. (My Api key and secret are saved in that file).
Do I need a new conf.js file?

**Comment by tuxitor** [comment]
I suspect you have an early version of zenbot 4. The conf.js is now supposed to look for conf.js in the main zenbot directory as _/Users/ron/zenbot/conf.js_
I suggest you upgrade. If you installed using _git clone_, the best method is to execute a _git pull_ from the _/Users/ron/zenbot directory_

**Comment by rondijkstra** [comment]
I have executed a _git pull_ as you suggested:

`RMBP:zenbot rondijkstra$ git pull
remote: Counting objects: 12, done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 12 (delta 8), reused 12 (delta 8), pack-reused 0
Unpacking objects: 100% (12/12), done.
From https://github.com/carlos8f/zenbot
   a036999..700968d  master     -> origin/master
Updating a036999..700968d
Fast-forward
 extensions/poloniex/exchange.js  | 18 ++++++++++++------
 extensions/trend_ema/strategy.js |  2 +-
 lib/engine.js                    | 13 ++++++-------
 3 files changed, 19 insertions(+), 14 deletions(-)
RMBP:zenbot rondijkstra$ git pull
Already up-to-date.
RMBP:zenbot rondijkstra$ zenbot trade poloniex.vtc-btc
fetching pre-roll data:

skipping 97 hrs of previously collected data

poloniex.VTC-BTC saved 200 trades 0 days left

download complete!
`

I see some files where updated. Do I need a restart after that or so?

**Comment by tuxitor** [comment]
Yes, you need to restart (after copying conf-sample.js and editing your cedentials)

**Comment by carlos8f** [comment]
> I suspect you have an early version of zenbot 4. The conf.js is now supposed to look for conf.js in the main zenbot directory as /Users/ron/zenbot/conf.js

yes, I moved extension confs into the main conf

**Comment by rondijkstra** [comment]
It keeps telling me I should enter my credentials... well here's my conf.js: 

I assume API and secret should do the job?

`var c = module.exports = {}

// mongo configuration
c.mongo = {}
c.mongo.host = 'localhost'
c.mongo.port = 27017
c.mongo.db = 'zenbot4'
c.mongo.username = null
c.mongo.password = null

// default selector. only used if omitting [selector] argument from a command.
c.selector = ‘poloniex.LTC-BTC'
// name of default trade strategy
c.strategy = 'trend_ema'

// Exchange API keys:

// to enable GDAX trading, enter your API credentials:
c.gdax = {}
c.gdax.key = 'YOUR-API-KEY'
c.gdax.b64secret = 'YOUR-BASE64-SECRET'
c.gdax.passphrase = 'YOUR-PASSPHRASE'

// to enable Poloniex trading, enter your API credentials:
c.poloniex = {}
**c.poloniex.key = 'KEYKEYKEYKEYK....'
c.poloniex.secret = 'SECRETSECRETSEC...'**

// Optional stop-order triggers:

// sell if price drops below this % of bought price (0 to disable)
c.sell_stop_pct = 0
// buy if price surges above this % of sold price (0 to disable)
c.buy_stop_pct = 0
// enable trailing sell stop when reaching this % profit (0 to disable)
c.profit_stop_enable_pct = 0
// maintain a trailing stop this % below the high-water mark of profit
c.profit_stop_pct = 1

// Order execution rules:

// avoid trading at a slippage above this pct
c.max_slippage_pct = 5
// buy with this % of currency balance
c.buy_pct = 99
// sell with this % of asset balance
c.sell_pct = 99
// ms to adjust non-filled order after
c.order_adjust_time = 30000
// avoid selling at a loss below this pct
c.max_sell_loss_pct = 25
// ms to poll order status
c.order_poll_time = 5000
// ms to wait for settlement (after an order cancel)
c.wait_for_settlement = 5000
// ms to wait for settlement (after a funds on hold error)
c.wait_more_for_settlement = 60000
// % to mark up or down price for orders
c.markup_pct = 0

// Misc options:

// default # days for backfill and sim commands
c.days = 14
// ms to poll new trades at
c.poll_trades = 30000
// amount of currency to start simulations with
c.currency_capital = 1000
// amount of asset to start simulations with
c.asset_capital = 0
// for sim, reverse time at the end of the graph, normalizing buy/hold to 0
c.symmetrical = false
// number of periods to calculate RSI at
c.rsi_periods = 14
// period to record balances for stats
c.balance_snapshot_period = '15m'
`

**Comment by rondijkstra** [comment]
/STOP!

I think it's working now. I have edited my conf.js again. and tried starting trading with `zenbot trade poloniex.VTC-BTC`.

What will be the conditions that zenbot really does the trading? I have no BTC left (since I already bought VTC the other day). Will Zenbot 'see' that I already own those coins and will start trading with those?

**Comment by rondijkstra** [comment]
This is what I have now. I don't see any trading takes place. 

<img width="697" alt="screen shot 2017-05-22 at 13 44 46" src="https://cloud.githubusercontent.com/assets/28846098/26308155/57d29958-3ef8-11e7-8bdc-45660429e6ff.png">

Am I missing a step?

**Comment by rondijkstra** [comment]
Solved! I had to add my IP address to the trusted IP addresses in Poloniex. After that the balance showed up (like the amount to coins I have and the amount of bitcoins).
